### PR TITLE
fix notice on __get error when no trace function/file is available (internal PHP functions/methods)

### PR DESCRIPTION
**Pull request by sakarikl**
If backtrace does not have file or line data error string generating caused notice

### CODE REVIEW

File: src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php [hunk 1]
```diff
@@ -108,8 +108,8 @@ private static function getUndefinedPropertyNotice(string $operationType, string
             . '            \'Undefined property: %s::$%s in %s on line %s\',' . "\n"
             . '            get_parent_class($this),' . "\n"
             . '            $' . $nameParameter . ',' . "\n"
-            . '            $backtrace[0][\'file\'],' . "\n"
-            . '            $backtrace[0][\'line\']' . "\n"
+            . '            $backtrace[0][\'file\'] ?? \'\',' . "\n"
```
Comment (Ocramius) [inline src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php]: In which scenarios would this be the case? Can you add a functional test (`.phpt`)?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php (hunk 1 above)
Comment (sakarikl) [inline src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php]: Code like this:

`
class Foo {
    private $value = 'Value';
}

class Bar  extends Foo{
}

$factory = new \ProxyManager\Factory\LazyLoadingValueHolderFactory();
$proxy = $factory->createProxy(
    \Bar::class,
    function (& $wrappedObject, $proxy, $method, $parameters, & $initializer) {
        $wrappedObject = new Bar(); // instantiation logic here
        $initializer   = null; // turning off further lazy initialization
    }
);

$r = new \ReflectionClass(\Foo::class);
$p = $r->getProperty('value');
$p->setAccessible(true);

$p->getValue($proxy);
`

causes output:

PHP Notice:  Undefined index: file in /vendor/ocramius/proxy-manager/src/ProxyManager/GeneratorStrategy/EvaluatingGeneratorStrategy.php(70) : eval()'d code on line 85

PHP Notice:  Undefined index: line in /vendor/ocramius/proxy-manager/src/ProxyManager/GeneratorStrategy/EvaluatingGeneratorStrategy.php(70) : eval()'d code on line 86

PHP Notice:  Undefined property: Bar::$value in  on line  in /vendor/ocramius/proxy-manager/src/ProxyManager/GeneratorStrategy/EvaluatingGeneratorStrategy.php(70) : eval()'d code on line 88

**Comment by Ocramius** [comment]
Ah yes, internal reflection produces no file/line in a trace, good point!

**Comment by Ocramius** [comment]
While trying to check for a suitable test case, I found that this actually hits a known edge case ( https://github.com/Ocramius/ProxyManager/pull/299 ). Currently unsure if fixable, since the code should work without failures in this scenario:

```php
class Foo {
    private $value = 'Value';
}

class Bar  extends Foo{
}

$factory = new \ProxyManager\Factory\LazyLoadingValueHolderFactory();
$proxy = $factory->createProxy(
    \Bar::class,
    function (& $wrappedObject, $proxy, $method, $parameters, & $initializer) {
        $wrappedObject = new Bar(); // instantiation logic here
        $initializer   = null; // turning off further lazy initialization
    }
);

$r = new \ReflectionClass(\Foo::class);
$p = $r->getProperty('value');
$p->setAccessible(true);

$p->getValue($proxy);
```

**Comment by Ocramius** [comment]
Indeed, I cannot reproduce this outside of this scenario (which is already broken).

Tried various proxy types, but this one seems to be the only affected one...
